src/graph/Graph.tsx

```diff
@@ -140,26 +146,37 @@ const GraphComponent = ({ data, config: cfg }: Props) => {
   }
 
   const onHighlightChange = (value: Highlight) => {
-    setGraph(resetHighlight)
+    setGraph(mergedGraph)
```

Comment:
These parts in a couple of places were meant to reset highlighting. 
The case with `mergedGraph` is only a particular highlight mode. I'd rather put a conditional here explicitly if you want to preserve highlights. But suppose you already have it somewhere when you compute `mergedGraph` 

```
if( highlights === "diffing") {
 setGraph(mergedGraph) { 
} else {
 setGraph(resetHighlights)
}
```